src/lib/scheduler/job_task.hpp

```diff
@@ -28,7 +28,9 @@ namespace opossum {
  */
 class JobTask : public AbstractTask {
  public:
-  explicit JobTask(const std::function<void()>& fn) : _fn(fn) {}
+  explicit JobTask(const std::function<void()>& fn, bool stealable = true) : AbstractTask(stealable), _fn(fn) {}
+
+  void schedule(NodeID preferred_node_id = CURRENT_NODE_ID, SchedulePriority priority = SchedulePriority::JobTask);
```

Comment:
> I assume that it is because of this default parameter? 

Exactly. When you construct the JobTask into a `shared_ptr<AbstractTask>`, the `AbstractTask::schedule` method will be called instead of `JobTask::Schedule`

Moving it to the task constructor sounds good though! I think the thought behind how it is now was, that a task itself does not really care for its priority, the priority only should come into play when scheduling the task.
But with the changes in here I think it is better that way. I'll look into it.